### ISSUE TITLE
Does not see the author in the .authors.yml.

### ISSUE THREAD

**Issue by askurashev** [squidfunk/mkdocs-material#5748]
### Bug description

I registered the configuration concerning the authors. However, when generating documentation, I get an error:
`ERROR - Blog post 'Blog/posts/page.md' author 'nickname' unknown, not listed in .authors.yml`.

mkdocs.yml
```
blog:
  authors: true
  authors_file: .authors.yml
```
.authors.yml
```
nickname:
  name: Surname
  description: Description
  avatar: .png URL
```

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
-

### Reproduction

-

### Steps to reproduce

-

### Browser

Chrome

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [ ] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting. However, there's no reproduction here. Please carefully read our bug reporting guide to learn how to create a reproduction, or we're not able to help you. Also, please see #5735

**Comment by askurashev** [comment]
Thanks for the answer.
I did some additional research. The problem was solved when I removed the line `authors_file: .authors.yml` from the configuration file `mkdocs.yml`.

Why, if it is present, does it give the error I voiced earlier?

**Comment by squidfunk** [comment]
Glad you could resolve it. The problem is likely that the location of the `.authors.yml` was not correct. As [our documentation](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#+blog.authors_file) states, it needs to be relative to `blog_dir`, so live at `docs/blog/.authors.yml`, if you didn't change `blog_dir`.

**Comment by askurashev** [comment]
No. Еhe file `.authors.yml ` was in the correct location. In order for the authors function to work, I did not move this file, but only deleted the line from the configuration.
There is a clear error in the code.

**Comment by Momotoculteur** [comment]
Hello,

i got the same error as @askurashev, see how to reproduce that error @squidfunk 

I got that error:
```
INFO    -  Building documentation...
INFO    -  Cleaning site directory
ERROR   -  Blog post 'blog/posts/init-new-terraform-project.md' author 'bmaurice' unknown, not listed in .authors.yml
```

With this mkdocs configuration : 
```yaml
- blog:
      enabled: true
      authors_file: .authors.yml
``` 

But my .authors.yml is correct 
```yaml
bmaurice:
    name: Bastien MAURICE
    description: Software Engineer
```
and saved here in ./blog/.authors.yml. Same for my posts, exemple : 
```md
---
date: 2023-01-31 
authors: [bmaurice]
description: >
  test maurice
categories:
  - Blog
tags:
  - terraform
---
this is my article
```

if i modify my mkdocs config file like this :
```yaml
- blog:
      enabled: true
``` 

Finally, with the **authors_file** property deleted, all is working perfectly  : 
```
INFO    -  Building documentation...
INFO    -  Cleaning site directory
INFO    -  Documentation built in 0.56 seconds
INFO    -  [02:14:54] Watching paths for changes: 'docs', 'mkdocs.yml'
INFO    -  [02:14:54] Serving on http://127.0.0.1:8000/
```

**Comment by squidfunk** [comment]
[I can only repeat myself](https://github.com/squidfunk/mkdocs-material/issues/5748#issuecomment-1648084421) 😉

**Comment by squidfunk** [comment]
Okay, so huge apologies to everybody experiencing this problem. It was our fault 🤦‍♂️ At some point, we changed the path from which the `.authors.yml` file is resolved from the. `blog_dir` to the `docs_dir`, with the following default:

https://github.com/squidfunk/mkdocs-material/blob/d005e85cdb46ac15cc34f40609e06014e2ece7ad/src/plugins/blog/config.py#L77

The `{blog}` placeholder is replaced with the `blog_dir`, which means that [if you set what our documentation suggested](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#+blog.authors_file):

``` yaml
plugins:
  - blog:
      authors_file: .authors.yml
```

... the file was expected to live at `docs/.authors.yml`. The rationale for this was to generalize authors so we can later add authors functionality to other plugins or the theme itself without tying it to the blog plugin. Again, I'm sorry this caused confusion, and I'll reopen this issue as a documentation problem, so we can fix it.

**Comment by Momotoculteur** [comment]
Hey ! 
Sorry i didn't see the 'reproduction bug' page, i'm discovering mkdocs and your theme just from several days. I will be more professional for the next bug reproduction. Thanks btw to took time and had a look at this problem without more details :)
Bastien

**Comment by squidfunk** [comment]
Corrected in a0921bcaf.
